### PR TITLE
Kie server S2I hierarchical test

### PR DESCRIPTION
**Pull request by sutaakar**

### CODE REVIEW

File: test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/s2i/KieServerS2iHierarchicalIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+        KieContainerResource container = containers.get(0);
+        assertThat(container).isNotNull();
+        assertThat(container.getContainerId()).isNotNull().isEqualTo(CONTAINER_ID);
+
+        ReleaseId containerReleaseId = container.getResolvedReleaseId();
+        assertThat(containerReleaseId).isNotNull();
+
+        SoftAssertions.assertSoftly(softly -> {
+            softly.assertThat(containerReleaseId.getGroupId()).isNotNull().isEqualTo(USERTASK.getGroupId());
```
Comment (jhrcek) [inline test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/s2i/KieServerS2iHierarchicalIntegrationTest.java]: isEqualTo's implementation calls isNotNull internally -> you can remove those calls. This comment applies across the entire PR where isNotNull is followed by another chained assetion in the same expression.

Comment (sutaakar) [reply test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/s2i/KieServerS2iHierarchicalIntegrationTest.java]: Good point, adjusted.

File: framework-cloud/framework-cloud-api/src/main/java/org/kie/cloud/api/settings/builder/KieServerS2ISettingsBuilder.java [hunk 2]
```diff
@@ -119,11 +119,22 @@
...
      */
     KieServerS2ISettingsBuilder withSourceLocation(String gitRepoUrl, String gitReference, String gitContextDir);
 
+    /**
+     * Return configured builder with Source location
+     *
+     * @param gitRepoUrl Repository url.
+     * @param gitReference Repository reference (branch/tag). E.g. 'master'.
+     * @param gitContextDir Repository context (location of pom file).
+     * @param artifactDirs Directories containing built kjars.
```
Comment (jakubschwan) [inline framework-cloud/framework-cloud-api/src/main/java/org/kie/cloud/api/settings/builder/KieServerS2ISettingsBuilder.java]: ARTIFACT_DIR accept list of artifacts, please add to doc in which format are artifacts accepted. (dir1,dir2,dir3)

**Comment by sutaakar** [comment]
retest this please

**Comment by sutaakar** [comment]
retest this please

**Pull request merged**